src/lib/tradingView/dydxfeed/utils.ts

```diff
@@ -0,0 +1,106 @@
+import BigNumber from 'bignumber.js';
+import { sum } from 'lodash';
+import { Mark, ResolutionString } from 'public/tradingview/charting_library';
+
+import { AbacusOrderSide, SubaccountFills } from '@/constants/abacus';
+import { RESOLUTION_TO_INTERVAL_MS } from '@/constants/candles';
+import { STRING_KEYS, StringGetterFunction, SupportedLocales } from '@/constants/localization';
+import { TOKEN_DECIMALS } from '@/constants/numbers';
+
+import { formatNumberOutput, OutputType } from '@/components/Output';
+
+import { RootStore } from '@/state/_store';
+import { getFillDetails } from '@/state/accountSelectors';
+
+/**
+ * @description Converts times in ms to the appropriate bar time (in seconds)
+ * For example, if the starting time = 5000ms, interval = 10,000ms, a value of 26,000ms would
+ * be grouped into the bar at 25,000ms = 25s
+ */
+export function getBarTime(
+  chartStartTimeMs: number,
+  fillTimeMs: number,
+  resolution: ResolutionString
+): number | undefined {
+  const intervalMs = RESOLUTION_TO_INTERVAL_MS[resolution];
+
+  const [startBn, intervalSizeBn, fillTimeBn] = [
+    BigNumber(chartStartTimeMs),
+    BigNumber(intervalMs),
+    BigNumber(fillTimeMs),
+  ];
+  const numIntervalsBetween = fillTimeBn.minus(startBn).dividedToIntegerBy(intervalSizeBn);
+  return startBn
+    .plus(numIntervalsBetween.multipliedBy(intervalSizeBn))
+    .dividedToIntegerBy(1000)
+    .toNumber();
+}
+
+function averageFillPrice(fills: SubaccountFills) {
```

Comment:
need to consolidate this method with logic from [this PR](https://github.com/dydxprotocol/v4-web/pull/921)